acme/acme/challenges.py

```diff
@@ -102,19 +100,73 @@ def port(self):
         """Port that the ACME client should be listening for validation."""
         return 443 if self.tls else 80
 
-    def uri(self, domain):
+    def uri(self, domain, chall):
         """Create an URI to the provisioned resource.
 
         Forms an URI to the HTTPS server provisioned resource
         (containing :attr:`~SimpleHTTP.token`).
 
         :param unicode domain: Domain name being verified.
+        :param challenges.SimpleHTTP chall:
 
         """
         return self._URI_TEMPLATE.format(
-            scheme=self.scheme, domain=domain, path=self.path)
+            scheme=self.scheme, domain=domain, token=chall.encode("token"))
```

Comment:
This seems like a mistake: isn't self.scheme going to be either the integer 443 or 80, according to line 101? So wouldn't formatting the URI template with that end up with something like

443://example.com/.well-known/acme-challenge/sometoken

? Shouldn't the scheme method return "http" or "https" instead of 443 or 80 for this use?